app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -41,17 +53,21 @@ function ContainerTopologyCtrl($scope, $http, $interval, $location, topologyServ
   $scope.legendTooltip = __("Click here to show/hide entities of this type");
 
   $scope.show_hide_names = function() {
-     var vertices = $scope.vs;
-
-     if ($scope.checkboxModel.value) {
-       vertices.selectAll("text.attached-label")
-         .classed("visible", true);
-     } else {
-       vertices.selectAll("text.attached-label")
-         .classed("visible", false);
-     }
+    checkboxValue = $('input#box_display_names')[0].checked
+    var vertices = $scope.vs;
+
+    $scope.checkboxModel.value = $()
+
+    if (checkboxValue) {
+      vertices.selectAll("text.attached-label")
+        .classed("visible", true);
+    } else {
+      vertices.selectAll("text.attached-label")
+        .classed("visible", false);
+    }
   };
 
+  $('input#box_display_names').click($scope.show_hide_names)
```

Comment:
the checkbox comes from the toolbar, 
I tried it with `{'data' => {'function' => 'miqCallAngular', 'function-data' => '{"name":"somefunc", "args":[]}'}}` but it didnt work...